oi_seattracker/management/commands/assign_participants.py

```diff
@@ -0,0 +1,26 @@
+from django.core.management.base import BaseCommand
+from oi_seattracker.models import Participant, Computer
+import json
+
+class Command(BaseCommand):
+    help = 'Assigns participants to their computers based on a json file (so-called "autoassign")'
+    
+    def add_arguments(self, parser):
+        parser.add_argument('source_file', type=open)
+
+    def handle(self, *args, **options):
+        data = json.load(options['source_file'])
+        mapping = {}
+        for line in data:
+            participant_id = line[0]
+            identifier = line[1]
+            mapping[participant_id] = identifier
+
+        participants = Participant.objects.all()
+        for participant in participants:
+            try:
+                computer = Computer.objects.get(ip_address=mapping[participant.pk])
+            except:
+                computer = Computer.objects.get(nice_name=mapping[participant.pk])
```

Comment:
The way you implemented it, if the input file doesn't assign all participants, you will get an exception, and, depending on the order in which `Participant.objects.all()` returns participants, only some of the participants will have their assignment set / changed.

Wouldn't it be easier to just iterate over the "lines" in the data, and `Participant.objects.get(pk=participant_id)` as you go?

Alternatively, you could improve the error handling, so that when `participant.pk` is not in `mapping`, you either skip that participant, possibly with a warning, or maybe do throw some kind of "some participants were missing" error, but at the end of the function, once all the matching participants have been processed?